google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/DefaultAppEngineAssetProvider.java

```diff
@@ -44,4 +46,25 @@ public XmlFile loadAppEngineStandardWebXml(@NotNull Project project, @NotNull Ar
 
     return null;
   }
+
+  @Nullable
+  @Override
+  public XmlFile loadAppEngineStandardWebXml(@NotNull Project project, @NotNull Module module) {
+    VirtualFile[] contentRoots = ModuleRootManager.getInstance(module).getContentRoots();
+
+    for (VirtualFile contentRoot : contentRoots) {
+      // TODO clean this up - do we need something more robust?
+      VirtualFile descriptorFile
+          = contentRoot.findFileByRelativePath("web/WEB-INF/appengine-web.xml");
```

Comment:
You could check out JavaPsiFacade, or have a look at ServerToIdeFileResolver.java in the debugger part.

Some more discussion on a similar topic in https://intellij-support.jetbrains.com/hc/en-us/community/posts/206101899-Getting-all-java-files-within-a-project?page=1#community_comment_206081095.